aiven_mysql_migrate/migration.py

```diff
@@ -336,10 +336,15 @@ def _reader_stderr(proc):
         return dump_processor.get_gtid()
 
     def _set_gtid(self, gtid: str):
-        LOGGER.info("Setting GTID in target database to `%s`", gtid)
+        LOGGER.info("GTID from the dump is `%s`", gtid)
 
         with self.target_master.cur() as cur:
-            cur.execute("SET @@GLOBAL.GTID_PURGED = %s", (gtid, ))
+            # Check which of the source GTIDs are not yet applied - needed in case of running migration again on top
+            # of finished one
+            cur.execute("SELECT GTID_SUBTRACT(%s, @@GTID_EXECUTED) AS DIFF", (gtid,))
+            new_gtids = "+" + cur.fetchone()["DIFF"]
```

Comment:
Can diff be empty? If yes then presumably need logic to skip trying to set it when that happens.

Reply:
If diff is empty then setting purged will be `= '+'` which is a noop, but it does make sense to skip this step and show the message in the log to make it more explicit.